widget/list.go

```diff
@@ -95,6 +95,19 @@ func (l *List) MinSize() fyne.Size {
 	return l.BaseWidget.MinSize()
 }
 
+// SetItemHeight supports changing the height of the specified list item. Items normally take the height of the template
+// returned from the CreateItem callback. The height parameter uses the same units as a fyne.Size type and refers
+// to the internal content height not including the divider size.
+//
+// Since: 2.3
+func (l *List) SetItemHeight(id ListItemID, height float32) {
+	if l.itemHeights == nil {
+		l.itemHeights = make(map[ListItemID]float32)
```

Comment:
I am a bit worried of this bit of code as it means every access to itemHeights could get access to a nil and they should all be protected. Not to sure how to address this, maybe having a function for all API entry point to check if this map is allocated and allocate it if not.

Reply:
It is safe to index, we always check `ok` to know the value is real.

The code:
```go
func main() {
	var m map[int]string

	v, ok := m[4]
	fmt.Println("Hello", v, ok)
}
```

Returns "Hello  false"

Reply:
Oh, I didn't know that. Good!